From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Thu, 22 Mar 2018 22:11:57 +0100
Subject: Battery API: return nothing

Stubs out the Battery API (https://developer.mozilla.org/docs/Web/API/Battery_Status_API) - Prevents websites from retrieving your device's battery charge.

(Includes @thestinger's fix for correct charging/unknown values)

Initial Commit: https://github.com/uazo/cromite/commit/776d4f859e5d6b20cb7874dad97e3d50c0f684ba#diff-9c226496fe0ea10334d4d7ea67998bec807fc780512c303a97bb78591c90a8f4 (Inherited from Bromite)
Category: Privacy
License: GPL-3.0-only - https://spdx.org/licenses/GPL-3.0-only.html
---
 .../modules/battery/battery_manager.cc        | 26 +++----------------
 1 file changed, 4 insertions(+), 22 deletions(-)

diff --git a/third_party/blink/renderer/modules/battery/battery_manager.cc b/third_party/blink/renderer/modules/battery/battery_manager.cc
--- a/third_party/blink/renderer/modules/battery/battery_manager.cc
+++ b/third_party/blink/renderer/modules/battery/battery_manager.cc
@@ -80,46 +80,28 @@ ScriptPromise<BatteryManager> BatteryManager::StartRequest(
 }
 
 bool BatteryManager::charging() {
-  return battery_status_.Charging();
+  return true;
 }
 
 double BatteryManager::chargingTime() {
-  return battery_status_.charging_time().InSecondsF();
+  return 0.0;
 }
 
 double BatteryManager::dischargingTime() {
-  return battery_status_.discharging_time().InSecondsF();
+  return std::numeric_limits<double>::infinity();
 }
 
 double BatteryManager::level() {
-  return battery_status_.Level();
+  return 1.0;
 }
 
 void BatteryManager::DidUpdateData() {
   DCHECK(battery_property_);
 
-  BatteryStatus old_status = battery_status_;
-  battery_status_ = *battery_dispatcher_->LatestData();
-
   if (battery_property_->GetState() == BatteryProperty::kPending) {
     battery_property_->Resolve(this);
     return;
   }
-
-  DCHECK(GetExecutionContext());
-  if (GetExecutionContext()->IsContextPaused() ||
-      GetExecutionContext()->IsContextDestroyed()) {
-    return;
-  }
-
-  if (battery_status_.Charging() != old_status.Charging())
-    DispatchEvent(*Event::Create(event_type_names::kChargingchange));
-  if (battery_status_.charging_time() != old_status.charging_time())
-    DispatchEvent(*Event::Create(event_type_names::kChargingtimechange));
-  if (battery_status_.discharging_time() != old_status.discharging_time())
-    DispatchEvent(*Event::Create(event_type_names::kDischargingtimechange));
-  if (battery_status_.Level() != old_status.Level())
-    DispatchEvent(*Event::Create(event_type_names::kLevelchange));
 }
 
 void BatteryManager::RegisterWithDispatcher() {
--
